File: 이것이 취업을 위한 코딩 테스트다/정렬/계수_정렬/계수_정렬.java
public class 계수_정렬 {
  public static void main(String[] args) {
    int[] arr = {7, 5, 9, 0, 3, 1, 6, 2, 9, 1, 4, 8, 0, 5, 2};
    int max = 9;
    int[] cnt = new int[max + 1];

    for (int num : arr) {
      cnt[num]++;
    }

    for (int i = 0; i <= max; i++) {
      for (int j = 0; j < cnt[i]; j++) {
        System.out.print(i + " ");
      }
    }
  }
}

File: 이것이 취업을 위한 코딩 테스트다/정렬/계수_정렬/계수_정렬.cpp
#include <iostream>
using namespace std;

int main() {
  int arr[] = {7, 5, 9, 0, 3, 1, 6, 2, 9, 1, 4, 8, 0, 5, 2};
  int size = sizeof(arr) / sizeof(arr[0]);
  int cnt[10] = {0};

  for (int i = 0; i < size; i++) {
    cnt[arr[i]]++;
  }

  for (int i = 0; i <= 9; i++) {
    for (int j = 0; j < cnt[i]; j++) {
      cout << i << ' ';
    }
  }

  return 0;
}
